drivers/QSPI.cpp

```diff
@@ -13,223 +13,229 @@
+        case QSPI_ADDR_SIZE_32:_address_size = QSPI_CFG_ADDR_SIZE_32; break;            
         default:_address_size = QSPI_CFG_ADDR_SIZE_8;
     }
     
     switch(alt_width) {
-        case 1:_alt_width = QSPI_CFG_BUS_SINGLE; break;
-        case 2:_alt_width = QSPI_CFG_BUS_DUAL; break;
-        case 4:_alt_width = QSPI_CFG_BUS_QUAD; break;            
+        case QSPI_BUS_SINGLE:_alt_width = QSPI_CFG_BUS_SINGLE; break;
+        case QSPI_BUS_DUAL:_alt_width = QSPI_CFG_BUS_DUAL; break;
+        case QSPI_BUS_QUAD:_alt_width = QSPI_CFG_BUS_QUAD; break;            
         default:_alt_width = QSPI_CFG_BUS_SINGLE;
     }
     
     switch(alt_size) {
-        case 0:_alt_size = QSPI_CFG_ALT_SIZE_NONE; break;
-        case 8:_alt_size = QSPI_CFG_ALT_SIZE_8; break;
-        case 16:_alt_size = QSPI_CFG_ALT_SIZE_16; break;
-        case 24:_alt_size = QSPI_CFG_ALT_SIZE_24; break;
-        case 32:_alt_size = QSPI_CFG_ALT_SIZE_32; break;            
+        case QSPI_ALT_SIZE_NONE:_alt_size = QSPI_CFG_ALT_SIZE_NONE; break;
+        case QSPI_ALT_SIZE_8:_alt_size = QSPI_CFG_ALT_SIZE_8; break;
+        case QSPI_ALT_SIZE_16:_alt_size = QSPI_CFG_ALT_SIZE_16; break;
+        case QSPI_ALT_SIZE_24:_alt_size = QSPI_CFG_ALT_SIZE_24; break;
+        case QSPI_ALT_SIZE_32:_alt_size = QSPI_CFG_ALT_SIZE_32; break;            
         default:_alt_size = QSPI_CFG_ALT_SIZE_NONE;
     }
     
     switch(data_width) {
-        case 1:_data_width = QSPI_CFG_BUS_SINGLE; break;
-        case 2:_data_width = QSPI_CFG_BUS_DUAL; break;
-        case 4:_data_width = QSPI_CFG_BUS_QUAD; break;            
+        case QSPI_BUS_SINGLE:_data_width = QSPI_CFG_BUS_SINGLE; break;
+        case QSPI_BUS_DUAL:_data_width = QSPI_CFG_BUS_DUAL; break;
+        case QSPI_BUS_QUAD:_data_width = QSPI_CFG_BUS_QUAD; break;            
         default:_data_width = QSPI_CFG_BUS_SINGLE; 
     }
     
     _num_dummy_cycles = dummy_cycles;
     _mode = mode;
     unlock();
     
-    return true;
+    return QSPI_SUCCESS;
 }
 
-bool QSPI::set_frequency(int hz) {
-    
+qspi_return_status_t QSPI::set_frequency(int hz) {
+    qspi_return_status_t ret_status = QSPI_SUCCESS; 
     lock();
     _hz = hz;
     
     //If the same owner, just change freq.
     //Otherwise we may have to change mode as well, so call _acquire
     if (_owner == this) {
-        qspi_frequency(&_qspi, _hz);
+        if(QSPI_STATUS_OK != qspi_frequency(&_qspi, _hz)) {
+            ret_status = QSPI_ERROR;
+        }
     } else {
         _acquire();
     }
     unlock();
     
-    return true;
+    return ret_status;
 }
 
-bool QSPI::initialize() {
+qspi_return_status_t QSPI::initialize() {
     lock();
     qspi_status_t ret = qspi_init(&_qspi, _qspi_io0, _qspi_io1, _qspi_io2, _qspi_io3, _qspi_clk, _qspi_cs, _hz, _mode );
     unlock();
     
-    return ( ret == QSPI_STATUS_OK )? true:false;
+    return ( ret == QSPI_STATUS_OK )? QSPI_SUCCESS:QSPI_ERROR;
```

Comment:
space `:` around, looks almost like `SPI_SUCCESS::QSPI_ERROR` . Looking at t his I really thought they is success that has error inside :)

Reply:
Will fix.